### PR TITLE
doc/cephfs/add-remove-mds: added cephadm note, refined "Adding an MDS"

### PR DESCRIPTION
**Pull request by dparmar18**
Description: 
1) Add a note about using cephadm for setting up the cluster and mds(s), mention 
    the use of ceph orchestrator if one needs to setup mds(s) manually.
2) Changed the term `data point` to `directory` in  point 1 under "Adding an MDS"
    section for better clarity.

Fixes: https://tracker.ceph.com/issues/54551

Signed-off-by: dparmar18 <dparmar@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/cephfs/add-remove-mds.rst [hunk 1]
```diff
@@ -62,7 +62,15 @@ means limiting its cache size.
...
-#. Create an mds data point ``/var/lib/ceph/mds/ceph-${id}``. The daemon only uses this directory to store its keyring.
+NOTE: It is highly recommended to use `cephadm`_ for setting up the ceph
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+cluster and thus the MDS(s). Use this approach only if you are setting up the
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ceph cluster manually. If one still intends to use the manual way for deploying
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+MDS daemons, `ceph orchestrator`_ can also be used.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#. Create an mds directory ``/var/lib/ceph/mds/ceph-${id}``. The daemon only uses this directory to store its keyring.
```
Comment (jtlayton) [inline doc/cephfs/add-remove-mds.rst:73]: Does the daemon not create this itself?

File: doc/cephfs/add-remove-mds.rst [hunk 2]
```diff
@@ -62,7 +62,15 @@ means limiting its cache size.
...
 
-#. Create an mds data point ``/var/lib/ceph/mds/ceph-${id}``. The daemon only uses this directory to store its keyring.
+NOTE: It is highly recommended to use `cephadm`_ for setting up the ceph
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+cluster and thus the MDS(s). Use this approach only if you are setting up the
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ceph cluster manually. If one still intends to use the manual way for deploying
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+MDS daemons, `ceph orchestrator`_ can also be used.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (jtlayton) [inline doc/cephfs/add-remove-mds.rst:72]: Question: do the carats (^) do some sort of markdown thing?

**Review by jtlayton — APPROVED** [review]
Content looks fine.

File: doc/cephfs/add-remove-mds.rst [hunk 3]
```diff
@@ -62,7 +62,15 @@ means limiting its cache size.
 Adding an MDS
 =============
 
-#. Create an mds data point ``/var/lib/ceph/mds/ceph-${id}``. The daemon only uses this directory to store its keyring.
+NOTE: It is highly recommended to use `cephadm`_ for setting up the ceph
```
Comment (batrick) [inline doc/cephfs/add-remove-mds.rst:65]: Correct markdown for this is:

`.. note:: the message`

See other places in docs for examples.

I think you should also update the introductory text of this doc to mention cephadm.

**Review by batrick — CHANGES_REQUESTED** [review]

File: doc/cephfs/add-remove-mds.rst (hunk 1 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst:73]: This is a doc that deals with manually adding the MDS. That's why you see this kind of instruction here.

File: doc/cephfs/add-remove-mds.rst (hunk 2 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst:72]: Yes, As it was decided that we need to provide a note that is fairly visible, using carats (^) over here would give us a bold text with a big font size that will be clearly visible. Although the size would be smaller than the section heading so it won't look odd at all.

File: doc/cephfs/add-remove-mds.rst (hunk 3 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst]: I've corrected the markdown for the `note`, thanks! 

I think you should also update the introductory text of this doc to mention cephadm.
> You mean to say I mention cephadm in the "DEPLOYING METADATA SERVERS" section?

File: doc/cephfs/add-remove-mds.rst [hunk 4]
```diff
@@ -62,7 +62,13 @@ means limiting its cache size.
 Adding an MDS
 =============
 
-#. Create an mds data point ``/var/lib/ceph/mds/ceph-${id}``. The daemon only uses this directory to store its keyring.
+.. note::
+   It is highly recommended to use `cephadm`_ for setting up the ceph cluster
```
Comment (batrick) [inline doc/cephfs/add-remove-mds.rst:66]:
```suggestion
   It is highly recommended to use `cephadm`_ or another Ceph orchestrator for setting up the ceph cluster
```

**Review by batrick — CHANGES_REQUESTED** [review]

File: doc/cephfs/add-remove-mds.rst (hunk 3 above)
Comment (batrick) [reply doc/cephfs/add-remove-mds.rst]: yes

File: doc/cephfs/add-remove-mds.rst (hunk 4 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst]: I've added the line, thanks for your input!

**Review by kotreshhr — APPROVED** [review]
lgtm

**Comment by vshankar** [comment]
jenkins render docs

**Comment by vshankar** [comment]
@dparmar18 please rebase and push

**Review by vshankar — APPROVED** [review]

**Review by batrick — APPROVED** [review]

File: doc/cephfs/add-remove-mds.rst [hunk 5]
```diff
@@ -61,8 +68,8 @@ means limiting its cache size.
 
 Adding an MDS
 =============
-
-#. Create an mds data point ``/var/lib/ceph/mds/ceph-${id}``. The daemon only uses this directory to store its keyring.
+
```
Comment (nmshelke) [inline doc/cephfs/add-remove-mds.rst:71]: Any reason for white-space? (I guess it is not intended)

File: doc/cephfs/add-remove-mds.rst [hunk 6]
```diff
@@ -1,3 +1,10 @@
+.. note::
+   It is highly recommended to use `cephadm`_ or another Ceph orchestrator for
+   setting up the ceph cluster and thus the MDS(s). Use this approach only if
+   you are setting up the ceph cluster manually. If one still intends to use
```
Comment (anthonyeleven) [inline doc/cephfs/add-remove-mds.rst:4]: s/ceph/Ceph/

File: doc/cephfs/add-remove-mds.rst [hunk 7]
```diff
@@ -1,3 +1,10 @@
+.. note::
+   It is highly recommended to use `cephadm`_ or another Ceph orchestrator for
+   setting up the ceph cluster and thus the MDS(s). Use this approach only if
```
Comment (anthonyeleven) [inline doc/cephfs/add-remove-mds.rst:3]: s/ceph/Ceph/

Also suggest removing " and thus the MDS(s)", I think it's reasonable to subsume them as part of the cluster within a CephFS document.

File: doc/cephfs/add-remove-mds.rst (hunk 5 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst:71]: I was trying to follow the style of formatting that I saw in the doc. If you see other headings too, there's a white space following them.

File: doc/cephfs/add-remove-mds.rst (hunk 7 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst]: I've removed it from the note, thanks!

File: doc/cephfs/add-remove-mds.rst (hunk 6 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst]: Sorry, I didn't understand, can you please elaborate on it?

File: doc/cephfs/add-remove-mds.rst [hunk 8]
```diff
@@ -107,3 +113,5 @@ the following method.
 	$ sudo rm -rf /var/lib/ceph/mds/ceph-${id}
 
 .. _MDS Config Reference: ../mds-config-ref
+.. _cephadm: ../../cephadm/index.html
+.. _ceph orchestrator: ../../cephadm/services/mds/
```
Comment (neesingh-rh) [inline doc/cephfs/add-remove-mds.rst:117]: Here also: s/ceph/Ceph

File: doc/cephfs/add-remove-mds.rst (hunk 6 above)
Comment (nmshelke) [reply doc/cephfs/add-remove-mds.rst]: Replace "ceph" with "Ceph"

File: doc/cephfs/add-remove-mds.rst (hunk 5 above)
Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst]: I've removed the unwanted white space

Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst]: I've removed the unwanted white space

File: doc/cephfs/add-remove-mds.rst (hunk 8 above)
Comment (joscollin) [reply doc/cephfs/add-remove-mds.rst:117]: drop this change and use `:ref:` instead.

**Review by joscollin — CHANGES_REQUESTED** [review]

File: doc/cephfs/add-remove-mds.rst (hunk 8 above)
Comment (neesingh-rh) [reply doc/cephfs/add-remove-mds.rst:117]: Yeah,that would be better:)

Comment (joscollin) [reply doc/cephfs/add-remove-mds.rst:117]: To help you, read: https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#cross-referencing-using-roles. `:doc:` could also be used.

Comment (dparmar18) [reply doc/cephfs/add-remove-mds.rst]: I've made the changes

**Review by joscollin — APPROVED** [review]

**Review by nmshelke — APPROVED** [review]
lgtm

**Pull request merged**